classes/local/skel/base.php

```diff
@@ -64,6 +64,15 @@ public function set_data(array $data) {
         $this->data = $data;
     }
 
+    public function set_attribute($attribute) {
```

Comment:
This could be used to replace things like $this->data['has_dependencies'] in version_php_file with $this->data['self']['has_dependencies']. We've talked about using the 'self' field for internal variables like this.

I've used it to enable the generation of the navigation functions and a function in grade.php.